hack/run-e2e-tests.sh

```diff
@@ -12,7 +12,7 @@ set -x
 go test \
     -test.short="${METERING_SHORT_TESTS}" \
     -test.v \
-    -timeout 20m \
+    -timeout 15m \
```

Comment:
any rhyme or reason, or just because it makes tests Go Faster™?

Reply:
No I lowered it so that it will timeout before jenkins times out, so our tests exit our scripts can run cleanup steps before jenkins kills the script which collects information. Plus our tests already run in under 15, so I'd like to keep it that way with these changes.

Reply:
Ok, if cleanup depends on this number being <20 we need to add a comment to that effect. Reasonable goal though.

Reply:
It's really arbitrary based on the environment running it, and only because the way jenkins kills the process doesn't let our scripts receive the signal apparently (might be due to how we run things). Once we move to origin CI, our script will get a proper signal if there's a timeout.